### ISSUE TITLE
There is no volume setting built in to the app

### ISSUE THREAD

**Issue by rdenno** [jellyfin/jellyfin-androidtv#867]
This makes it impossible to change the volume when using Bluetooth audio on devices without built in volume controls. I apologize if this is not supposed to be here

**Comment by nielsvanvelzen** [comment]
I'm wondering what kind of device you're using. All Android devices support volume control and I have never seen volume controls in an app before.

**Comment by MovieWiz** [comment]
That may be true on mobile devices, @nielsvanvelzen.  But there is no "global" volume control on the FireStick and we use the AndroidTV app on the firestick.  So, if you connect bluetooth headphones or speakers to the firestick, there is no way in the app to change the volume.

VLC For Android and Plex For Android both support controlling the volume from within the app.

**Comment by mscroggi** [comment]
IMO this is a failure of the firestick... The volume buttons on the firestick remote should control the playback device volume controls, not the app..  I use Chromecast TV and the volume controls instruct the BT speaker to increase or decrease the volume.

**Comment by MovieWiz** [comment]
I'm not really advocating for the feature here.  Just trying to comment on rdenno's suggestion constructively.  I should point out to @mscroggi  that there are no volume control buttons on the FireStick Lite remote.  I completely concede that people could consider that a failure of the firestick.  But it is nevertheless reality.

I can control volume for my bluetooth headphones from the buttons on the headphones.  However, for someone using bluetooth speakers, this very well could be a physical action (i.e. you might have to go over to the speaker and adjust a knob) where this would be more conveniently handled within the app using the android device's remote control.

I must admit though that this is a pretty niche need.

**Comment by mscroggi** [comment]
AH!  I did not realize the FireStick Lite didn't have volume buttons..  I have a FireTV stick and it has the volume buttons.   It seems odd if a device is going to support bluetooth it doesn't have volume buttons..   interesting..  I guess it was to keep the price down for that model.

**Comment by mscroggi** [comment]
@MovieWiz I was curious about the BT controls with my FTV 4k.  I paired my speaker, but noticed even with the volume buttons, there is no way to control the volume at the speaker.   This is a big limitation for the FTV sticks.  I am really surprised at this.  This feature works flawlessly with my Chromecast TV.  I assumed if a device supported BT, it would include control features..  but that is not the case. Evidently there is a third party app “Precise Volume” that can be used to dial up and down the volumes on the FTV sticks.  But native support is not there..  very disappointing.

**Comment by mscroggi** [comment]
@MovieWiz  I just saw a note on the aftvnews.com that the firetv's were updated to control volume via BT -  https://www.aftvnews.com/fire-tvs-can-now-control-bluetooth-headphone-speaker-volume-using-the-remote-buttons

**Comment by Phoenix-004** [comment]
This is an important feature to have. And something you won't appreciate until you have used it.

This way, you can have your TV master volume set at one level, and control jellyfin media volume from within the app. Nothing more annoying than watching a movie or show, having to raise your master volume because you cant hear, switching to the youtube app or Netflix and then its too loud.

**Comment by nielsvanvelzen** [comment]
If we were to add volume control to the app it would change the volume of the system, not for the app individually. That's just the way sound mixing in Android works.

**Comment by Phoenix-004** [comment]
> If we were to add volume control to the app it would change the volume of the system, not for the app individually. That's just the way sound mixing in Android works.

I don't think that is true. When using Kodi on my android TV, I can change the volume just within the app. That is the single reason I use Kodi til this day and have to use the Jellyfin plugin in it.

Its simply a quality of life feature that would be great to have. I use my (85")TV with a sound bar connected, for my PC, PS5 and jellyfish. I can go into Jellyfin to play something, won't be able to hear the sound clearly, increase the volume, then switch to my PS5, and then its to loud, or to my PC.

Its annoyng.

**Comment by Lucide** [comment]
Hello, one way to look at this feature request is that (since what, 11 years?) fire “stick” devices have relied on the hdmi sink to handle volume. For an external "smart tv", that’s not unreasonable; as with most set-top boxes, duplicate master volume controls get messy (think of a hdmi bluray player with its own volume control, ouch).
As mentioned above, at some point the sticks enabled internal volume control for bluetooth devices, but different use case.

For dongles, deeper integration with a “dumb” panel is better, and that’s why smart tv-oriented players tends to offer in-player volume.

This is a point I hadn’t seen made in favour of an option for "internal volume".
Thank you for your work regardless 🙂.
Personally, we rely on "realtime" volume control to compensate for the limited dynamic range of our salvaged stereo speakers. We are students, so tight finances for audio equipment. This client is cool because I believe it's the only one which allows to force server-side downmixing.

**Comment by karoltheguy** [comment]
+1 on volume control
Somehow Jellyfin is the most quiet app I use on my tv.

**Comment by dbono8888** [comment]
+1 I know users with low volume issues that can't be resolved solely by their device.  The abilty to better manage & boost the volume of the client would be beneficial.
